__test__/routes/route-free-trial-request.test.js

```diff
@@ -1,32 +1,92 @@
 const request = require('superagent');
+const mongoose = require('mongoose');
+
+mongoose.Promise = require('bluebird');
+
+require('dotenv').config({ path: `${__dirname}/../test.env` });
+
 const server = require('./../../lib/server');
+const Student = require('./../../schemas/student');
+const sample1 = require('./../sample/samplePost.json');
+const sample2 = require('./../sample/samplePostMissingAddress');
+const sample3 = require('./../sample/samplePostMissingHasIntrument.json');
+const sample4 = require('./../sample/samplePostMissingAvailability.json');
 require('jest');
 
 describe('Testing free-trial-request route', () => {
-  beforeAll(() => {
-    process.env.PORT = 3000;
-    server.start();
-  });
+  const testIdsArray = [];
+  beforeAll(() =>
+    server.start().then(tests => {
+      console.log('Server has started in the tests');
+      return tests;
+    }));
   afterAll(() => {
-    server.stop();
+    testIdsArray.map(id =>
+      Student.findOneAndDelete({ _id: id }).catch(err => console.error(err))
+    );
+    return server
+      .stop()
+      .then(() => mongoose.disconnect())
+      .catch(err => console.error(err));
   });
 
   test('Making a POST call with no body will return error', () =>
     request
-      .post(':3000/api/free-trial-request')
-      .send()
+      .post(':5000/api/free-trial-request')
```

Comment:
I would suggest trying to use `process.env.PORT` instead of hard-coding the port number. Fewer maintainability issues and better consistency across the codebase.

Reply:
That makes a ton of sense.  Thanks!